### PR TITLE
dnf: add support for update_cache

### PR DESCRIPTION
**Pull request by mhatina**
Requires: https://github.com/rpm-software-management/dnf/pull/721

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
dnf module

##### SUMMARY
add support for missing 'update_cache' option

### CODE REVIEW

File: lib/ansible/modules/packaging/os/dnf.py [hunk 1]
```diff
@@ -316,6 +316,9 @@ def ensure(module, base, state, names):
     # and fail with a message about what they can't.
     failures = []
     allow_erasing = False
+    if module.params.get('update_cache') and state in ['installed', 'present', 'latest']:
```
Comment (robinro) [inline lib/ansible/modules/packaging/os/dnf.py]: Why require the state?
apt and zypper allow `update_cache` without any state and I don't see how dnf is different in that regard.

**Comment by robinro** [comment]
If I understand correctly this needs a really new (unreleased?) dnf version, so this should be noted in the docs and you probably want a try/except to not show a traceback on older dnf versions.

**Comment by igor-raits** [comment]
@robinro yes, same as for #20923

**Comment by robinro** [comment]
@mhatina thanks for the change. I think the try/except is sufficient, though it's the call of @ignatenkobrain and @DJMuggs.

I'd recommend to
* put a note in the documentation of update_cache, which dnf version is needed
* mention the version also in the error message

One can think about whether or not this should fail. For update_cache I'd rather not fail but show a warning and still install/remove that package. This is also done in the git module for features, which the given git version does not support.

**Pull request closed without merging**